src/components/Channel/Channel.js

```diff
@@ -1,148 +1,674 @@
+      attachments,
+      created_at: new Date(),
+      __html: text,
+      html: text,
+      id: `${client.userID}-${uuidv4()}`,
+      mentioned_users,
+      reactions: [],
+      status: 'sending',
+      text,
+      type: 'regular',
+      user: {
+        id: client.userID,
+        ...client.user,
+      },
+      ...extraFields,
+    };
+
+    if (parent?.id) {
+      message.parent_id = parent.id;
+    }
+    return message;
+  };
+
+  const sendMessageRequest = async (message) => {
+    const {
+      attachments,
+      created_at,
+      __html,
+      html,
+      id,
+      mentioned_users,
+      parent_id,
+      reactions,
+      status,
+      text,
+      type,
+      user,
+      ...extraFields
+    } = message;
+
+    const messageData = {
+      attachments,
+      id,
+      mentioned_users,
+      parent_id,
+      text,
+      ...extraFields,
+    };
+
+    try {
+      let messageResponse;
+
+      if (props.doSendMessageRequest) {
+        messageResponse = await props.doSendMessageRequest(
+          channel.cid,
+          messageData,
+        );
+      } else {
+        messageResponse = await channel.sendMessage(messageData);
+      }
+
+      if (messageResponse.message) {
+        messageResponse.message.status = 'received';
+        updateMessage(messageResponse.message);
+      }
+    } catch (error) {
+      console.log(error);
+      message.status = 'failed';
+      updateMessage(message);
+    }
+  };
+
+  const sendMessage = async ({
+    attachments = [],
+    mentioned_users,
+    parent,
+    text,
+    ...extraFields
+  }) => {
+    channel.state.filterErrorMessages();
+
+    const messagePreview = createMessagePreview({
+      attachments,
+      extraFields,
+      mentioned_users,
+      parent,
+      text,
+    });
+
+    updateMessage(messagePreview, {
+      commands: [],
+      messageInput: '',
+      userAutocomplete: [],
+    });
+
+    await sendMessageRequest(messagePreview);
+  };
+
+  const retrySendMessage = async (message) => {
+    message = message.asMutable();
+    message.status = 'sending';
+    updateMessage(message);
+    await sendMessageRequest(message);
+  };
+
+  const loadMoreFinished = (updatedHasMore, messages) => {
+    if (unmounted) return;
+    setLoadingMore(false);
+    setHasMore(updatedHasMore);
+    setMessages(messages);
+  };
+
+  // hard limit to prevent you from scrolling faster than 1 page per 2 seconds
+  const loadMoreFinishedDebounced = debounce(loadMoreFinished, 2000, {
+    leading: true,
+    trailing: true,
+  });
+
+  const loadMore = async () => {
+    if (loadingMore || hasMore === false || unmounted) return;
+    setLoadingMore(true);
+
+    if (!messages.length) {
+      return setLoadingMore(false);
+    }
+
+    const oldestMessage = messages[0] ? messages[0] : null;
+
+    if (oldestMessage?.status !== 'received') {
+      return setLoadingMore(false);
+    }
+
+    const oldestID = oldestMessage ? oldestMessage.id : null;
```

Comment:
are we needing null vs undefined for a reason?
```suggestion
    const oldestID = oldestMessage?.id;
```

Reply:
very interesting, just looked into the stream-chat-js code and it's expecting either a number or undefined for the pagination options, looks like we're currently giving either a string ID or null. so undefined is actually correct, but not sure if passing an message ID string here is working properly